pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -0,0 +1,277 @@
+)
+
+type Controller struct {
+	ctx                   context.Context
+	clusterClient         v3.ClusterInterface
+	clusterLister         v3.ClusterLister
+	backupClient          v3.EtcdBackupInterface
+	backupLister          v3.EtcdBackupLister
+	backupDriver          service.EngineService
+	KontainerDriverLister v3.KontainerDriverLister
+}
+
+func Register(ctx context.Context, management *config.ManagementContext) {
+	c := &Controller{
+		ctx:                   ctx,
+		clusterClient:         management.Management.Clusters(""),
+		clusterLister:         management.Management.Clusters("").Controller().Lister(),
+		backupClient:          management.Management.EtcdBackups(""),
+		backupLister:          management.Management.EtcdBackups("").Controller().Lister(),
+		backupDriver:          service.NewEngineService(clusterprovisioner.NewPersistentStore(management.Core.Namespaces(""), management.Core)),
+		KontainerDriverLister: management.Management.KontainerDrivers("").Controller().Lister(),
+	}
+
+	local := &rkedialerfactory.RKEDialerFactory{
+		Factory: management.Dialer,
+	}
+	docker := &rkedialerfactory.RKEDialerFactory{
+		Factory: management.Dialer,
+		Docker:  true,
+	}
+	driver := service.Drivers[service.RancherKubernetesEngineDriverName]
+	rkeDriver := driver.(*rke.Driver)
+	rkeDriver.DockerDialer = docker.Build
+	rkeDriver.LocalDialer = local.Build
+	rkeDriver.WrapTransportFactory = docker.WrapTransport
+
+	c.backupClient.AddLifecycle(ctx, "etcdbackup-controller", c)
+	go c.clusterBackupSync(ctx, clusterBackupCheckInterval)
+}
+
+func (c *Controller) Create(b *v3.EtcdBackup) (runtime.Object, error) {
+	cluster, err := c.clusterClient.Get(b.ClusterID, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	v3.BackupConditionCreated.True(b)
+	b.Filename = getBackupFilename(b.Name, cluster)
+	b.BackupConfig = *cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig
+	b, err = c.backupClient.Update(b)
+	if err != nil {
+		return nil, err
+	}
+
+	kontainerDriver, err := c.KontainerDriverLister.Get("", service.RancherKubernetesEngineDriverName)
+	if err != nil {
+		return nil, err
+	}
+	err = c.backupDriver.ETCDSave(context.Background(), cluster.Name, kontainerDriver, cluster.Spec, b.Name)
+	if err != nil {
+
+		return b, err
+	}
+	v3.BackupConditionCompleted.True(b)
+	return c.backupClient.Update(b)
+}
+
+func (c *Controller) Remove(b *v3.EtcdBackup) (runtime.Object, error) {
+	logrus.Infof("[etcd-backup] Deleteing backup %s ", b.Name)
+	if err := deleteS3Snapshot(b); err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (c *Controller) Updated(b *v3.EtcdBackup) (runtime.Object, error) {
+	return b, nil
+}
+
+func (c *Controller) clusterBackupSync(ctx context.Context, interval time.Duration) error {
+	for range ticker.Context(ctx, interval) {
+		clusters, err := c.clusterLister.List("", labels.NewSelector())
+		if err != nil {
+			logrus.Error(err)
+			return err
```

Comment:
since this is running in goroutine, will return error be displayed anywhere? 
if not, need to add additional info like function name to `logrus.Errorf`